js/src/forum/components/CategoriesPage.tsx

```diff
@@ -86,15 +106,21 @@ export default class CategoriesPage extends Page {
       <div className="CategoriesPage-content sideNavOffset">
         <ol className="TagCategoryList">
           {pinned.map((tag) => {
-            return Category.component({ model: tag });
+            return Category.component({ model: tag, enablePrimaryTagColor: app.forum.attribute('categories.enablePrimaryTagColor'), enablePrimaryChildTagColor: app.forum.attribute('categories.enablePrimaryChildTagColor') });
           })}
         </ol>
 
         {cloud.length ? <div className="TagCloud">{cloud.map((tag) => [tagLabel(tag, { link: true }), ' '])}</div> : ''}
       </div>,
       50
     );
-
+    // Only check for widget for the right if enable in the settings
+    if (app.forum.attribute('categories.widgetRight')) {
```

Comment:
Is there no way for the widgets plugin to provide an API for reliably getting it's widgets? Or for it to extend this plugin?

Reply:
Well the plugin doesn't have option to hide or show widget on pages, I've provided a toggle to give admin the choice of showing what is needed for their categories page. Some people have them as their main page, while I don't have it as the mainpage. But the widget is useful else where on the website.